src/components/Publish/_utils.ts

```diff
@@ -506,16 +507,53 @@ export async function verifyRawServiceCredential(
       }
     }
     if (response?.status === 201) {
-      const serviceOffering = parsedServiceCredential.verifiableCredential.find(
-        (credential) =>
-          credential?.credentialSubject?.type === 'gx:ServiceOffering'
-      )
-      const credentialId = serviceOffering?.credentialSubject?.id
-
-      return {
-        verified: true,
-        complianceApiVersion,
-        idMatch: did && did?.toLowerCase() === credentialId?.toLowerCase()
+      const serviceOfferings =
+        parsedServiceCredential.verifiableCredential.filter(
+          (credential) =>
+            credential?.credentialSubject?.type === 'gx:ServiceOffering'
+        )
+      if (serviceOfferings.length === 1) {
+        return {
+          verified: true,
+          complianceApiVersion,
```

Comment:
suggestion(blocking): this can be a local variable for the api success case as it is included in every return statement